subprojects/gradle/lint-report/src/main/kotlin/com/avito/android/lint/slack/LintReportToSlackTaskFactory.kt

```diff
@@ -0,0 +1,63 @@
+package com.avito.android.lint.slack
+
+import com.avito.android.lint.AndroidLintAccessor
+import com.avito.android.lint.LintReportExtension
+import com.avito.android.lint.validInGradleTaskName
+import com.avito.kotlin.dsl.dependencyOn
+import com.avito.kotlin.dsl.typedNamedOrNull
+import com.avito.logger.Logger
+import com.avito.slack.SlackClient
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.register
+
+class LintReportToSlackTaskFactory(
+    private val project: Project,
+    private val logger: Logger,
+    private val androidLintAccessor: AndroidLintAccessor = AndroidLintAccessor(project)
+) {
+
+    @Suppress("UnstableApiUsage")
+    private val slackClientProvider: Provider<SlackClient> by lazy {
+        val extension = project.extensions.getByType<LintReportExtension>()
+
+        extension.slackToken.zip(extension.slackWorkspace) { token, workspace ->
+            SlackClient.Impl(
+                token = token,
+                workspace = workspace
+            ) as SlackClient
+        }
+    }
+
+    /**
+     * To be used in CiStep, because slack channel only known from there
+     */
+    fun registerLintReportToSlackTask(slackChannel: String): TaskProvider<LintSlackReportTask> {
```

Comment:
Explain please, why don't we just send the same report from one task.
To make each interaction independent (loggable, cacheable, ...)?

Reply:
previously we had single task in graph for main app, like `:app:lintReportToSlack` it was configured multiple times from different `builds` in CiSteps plugin;

`fullCheck` configures and sets input property `slackChannel` in `:app:lintReportToSlack` to `#android-dev`
`regress` configures and sets input property `slackChannel` in `:app:lintReportToSlack` to `""`

now `fullCheck` configuration is broken and property set to `""`

Reply:
Initially, I thought about multiple destinations in one build.
Thanks, I see the conflict between different builds.
It seems we need to add configuration tests to address this problem in all steps.